### ISSUE TITLE
mouse is detected but the cursor is not moving, M510, Ubuntu 16.04, ID 046d:c52b

### ISSUE THREAD

**Issue by Motorrat** [pwr/Solaar#353]
GUI widget is working fine, I can pair and unpair the mouse. It is recognized and its battery status is reported when I move it or click a button. However nothing happens afterwards, the cursor is not moiving. 
It's a Dell XPS 13 laptop with two USB3 ports.
I have tried to connect the unifying reciever into each port directly, through a USB extention cord, and through a KVMP Switch (that way the receiver-mouse combination works with another Ubuntu 12.10 box and Windows 7 Dell laptop)

What else could I check?

Below is the dmesg snippet  
```
[139069.262159] bluetooth hci0:20 enx9cb6d0da8994: renamed from bnep0
[141236.754407] bluetooth hci0:4 enx9cb6d0da8994: renamed from bnep0
[148798.632871] usb 1-1: new full-speed USB device number 4 using xhci_hcd
[148798.776470] usb 1-1: New USB device found, idVendor=046d, idProduct=c52b
[148798.776476] usb 1-1: New USB device strings: Mfr=1, Product=2, SerialNumber=0
[148798.776479] usb 1-1: Product: USB Receiver
[148798.776482] usb 1-1: Manufacturer: Logitech
[148799.318131] usbcore: registered new interface driver usbhid
[148799.318133] usbhid: USB HID core driver
[148799.324190] logitech-djreceiver 0003:046D:C52B.0004: hiddev0,hidraw1: USB HID v1.11 Device [Logitech USB Receiver] on usb-0000:00:14.0-1/input2
[148799.458192] input: Logitech M510 as /devices/pci0000:00/0000:00:14.0/usb1/1-1/1-1:1.2/0003:046D:C52B.0004/0003:046D:1025.0005/input/input88
```

**Comment by jrbenito** [comment]
Hi @Motorrat ,

In fact Solaar does not play any role in regarding mouse movements or clicks, these events are interpreted by kernel drivers. 

Please, try to run `xev` and see if it reports anything about your mouse. Do you have another Unifying device to test against? Also, verify if your system is recognizing the mouse as valid input, as per your `dmesg` snippet there is nothing wrong at kernel side. It smells like X server not using input88.

Anyway, I have M510 that works out of the box on Ubuntu 16.10 (desktop computer). I just received my new dell laptop yesterday (although not a XPS 13) and will install Linux on it today. I will try my M510 and let you know.

BR,
Benito.

**Comment by Motorrat** [comment]
Hi @jrbenito, thanks for the tips! 
I have checked and xev doesn't show anything. I have three unified receivers and they all exibit the same behavior. 
What exactly could I check about the X server not using input88? Could it be any specific config file? Or is there a log I could check?
Thank you in advance for the guidance here!

**Comment by jrbenito** [comment]
Hi @Motorrat ,

Are you sure this mouse is working? Did you test it on Windows?

Regarding config, I don ´t know, never had this issue before, in the past you have to configure inputs manually to x server but this has gone for years on modern distributions.

points to check:

1. be sure mouse works
2. if 1, then test another linux distribution (if possible)
3. what modules are up? (lsmod) Are logitech ones up? This mouse is handled by hid with no special driver, it shall work on linux as soon as it is recognized.
4. input88 you mention, is it the mouse for sure?

BRs

**Comment by Motorrat** [comment]
Yes, the mouse is definitely OK, is used with the same receiver on Windows. Checked with the Ubuntu 12.10 and the receiver comes up in the dmesg, but mouse is not moving either.

**Comment by Motorrat** [comment]
Getting closer to the root cause:
```
[209096.915] (II) config/udev: Adding input device Logitech M510 (/dev/input/mouse0)
[209096.915] (II) No input driver specified, ignoring this device.
[209096.915] (II) This device may have been added with another device file.
[209096.969] (II) config/udev: Adding input device Logitech M510 (/dev/input/event6)
[209096.969] (**) Logitech M510: Applying InputClass "evdev pointer catchall"
[209096.969] (**) Logitech M510: Applying InputClass "natural scrolling for mouse wheel"
[209096.969] (**) Logitech M510: Applying InputClass "libinput pointer catchall"
[209096.969] (II) Using input driver 'libinput' for 'Logitech M510'
[209096.969] (**) Logitech M510: always reports core events
[209096.969] (**) Option "Device" "/dev/input/mice"
[209096.970] (**) Option "_source" "server/udev"
[209097.036] (II) failed to create input device '/dev/input/mice'.
[209097.036] (EE) libinput: Logitech M510: Failed to create a device for /dev/input/mice
[209097.036] (EE) PreInit returned 2 for "Logitech M510"
[209097.036] (II) UnloadModule: "libinput"
```

**Comment by Motorrat** [comment]
The problem was with a unnecessary and erroneous configuration file. All works now with that file removed.  https://askubuntu.com/questions/905269/logitech-unified-receiver-mouse-m510-ee-failed-to-load-module-mouse-module/905278#905278

**Comment by jrbenito** [comment]
Nice!

Em 15 de abr de 2017 19:25, "Diego Kobylkin" <notifications@github.com>
escreveu:

**Comment by upalatucci** [comment]
Hei guys,
Just writing for people in the future that will encounter the same issue.
Yesterday i had this issue with my logitech mx master mouse and solaar.
Cursor not moving but responsive to right and left clicks. 
The real problem was that the thumb button was broken!!
Couldn't hear the thumb button click!! I discover it's a widespread issue in these kinds of mice. 
From the OS viewpoint, the button was always down and this have triggered some solaar conditions i think. 
Also in windows, I have actions triggered by a combination of thumb button and mouse gesture.

TLDR: if you have this issue, check also that your thumb button is working properly
